main.cpp

```diff
@@ -278,12 +278,13 @@ int main (int argc, char** argv)
 				std::cerr << "Error dispatching: " << retval << std::endl;
 			else if (retval != 0)
 				packets_read = true;
+			else
```

Comment:
Still it's not a very clean solution I think.
May be in the future, a better redesign of the code will fix this kind of logic.